frontend/src/Components/Helpers/helpers.js

```diff
@@ -77,16 +77,11 @@ export async function checkFlowerTaskStatus (task_id) {
 
 export async function exponentialBackoff (checkStatus, task_id, timeout, max, delay, callback) {
     let status = await checkStatus(task_id);
-    if (status === "SUCCESS") {
-        callback();
-    } else {
-        if(status === "FAILURE" || max === 0){
-            clearTimeout(timeout);
-            return "FAILURE";
-        } else if (status === "PENDING") {
-            timeout = setTimeout(function() {
-                return exponentialBackoff(checkStatus, task_id, timeout, --max, delay * 2, callback);
-            }, delay);
-        } 
+    if (status === "SUCCESS" || status === "FAILURE" || max === 0) {
```

Comment:
You are already checking the status value here, but then you pass it over to the callback function and check it again. 
I think a simpler pattern would be to have the this function accept a `successCallback` and a `failureCallback` instead of a general callback.